sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -102,6 +103,32 @@ public void stop() throws ConnectException {
 		}
 	}
 
+	@Override
+	public Map<TopicPartition, OffsetAndMetadata> preCommit(
+		Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
+		Map<TopicPartition, OffsetAndMetadata> result = new HashMap<>();
+
+		log.info("{} performing preCommit", name());
+		log.info("{} offsets: {}", name(), currentOffsets);
+		log.info("{} compressed_file_size: {}", name(), GZIPThreshold);
+
+		partitions.entrySet().stream()
+			.filter(e -> e.getValue() != null && e.getValue().getWriter() != null && e.getValue().getWriter().getTotalCompressedSize() > GZIPThreshold)
```

Comment:
Are these not-null checks necessary? Is it possible that a `partitions` entry has a null value or it's a non-null but doesn't have a writer?